http-server/src/main/java/org/triplea/server/access/RoleAuthorizer.java

```diff
@@ -9,27 +9,38 @@
  */
 public class RoleAuthorizer implements Authorizer<AuthenticatedUser> {
 
-  /**
-   * Verifies a user is authorized to assume a requestedRole. <br>
-   * Rules are:<br>
-   * - admin is authorized all roles<br>
-   * - moderator is authorized all roles except admin<br>
-   * - player can access player or anonymous resources<br>
-   * - anonymous has lowest privilege and can access only the anonymous role
-   */
+  /** Verifies a user is authorized to assume a requestedRole. <br> */
   @Override
   public boolean authorize(final AuthenticatedUser user, final String requestedRole) {
     switch (user.getUserRole()) {
       case UserRole.ADMIN:
-        return true;
+        return adminPermissions(requestedRole);
       case UserRole.MODERATOR:
-        return !requestedRole.equals(UserRole.ADMIN);
+        return moderatorPermission(requestedRole);
       case UserRole.PLAYER:
-        return requestedRole.equals(UserRole.PLAYER) || requestedRole.equals(UserRole.ANONYMOUS);
+        return playerPermissions(requestedRole);
       case UserRole.ANONYMOUS:
-        return requestedRole.equals(UserRole.ANONYMOUS);
+        return anonymousPermissions(requestedRole);
+      case UserRole.HOST:
+        return requestedRole.equals(UserRole.HOST);
       default:
         throw new AssertionError("Unrecognized user role: " + user.getUserRole());
     }
   }
+
+  private static boolean adminPermissions(final String requestedRole) {
```

Comment:
`adminPermissions` is a confusing name in my opinion.
Without looking at the method above I'd actually think this method returns true if the passed role has admin permissions, instead of if the admin is authorized to perform actions for the specified role.
So I'd suggest renaming the methods to `isAdminAuthorizedFor` or something. Can't really think of something better right now, haven't convinced myself yet 😅

Reply:
It does read better with: `adminAuthorizedFor`. The 'is' prefix IMO gets to be a bit wordy.

359135935